app/models/category.rb

```diff
@@ -1,2 +1,4 @@
 class Category < ApplicationRecord
+  has_one :transaction_record, class_name: "Transaction"
```

Comment:
`has_one: :transaction` is fine. Also I don't think you will need to use `category.transaction`

Reply:
Also db and model level validations are missed.